### PR TITLE
model: Add qwen-omni models

### PR DESCRIPTION
**Pull request by Samoed**
Close https://github.com/embeddings-benchmark/mteb/issues/3985

I've got `0.00977` on `AudioSetStrongA2TRetrieval` with `Qwen/Qwen2.5-Omni-3B`

### CODE REVIEW

File: mteb/models/model_implementations/qwen_omni_lm.py [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+    primaryClass={cs.CL},
+    url={https://arxiv.org/abs/2509.17765},
+}""",
+)
+
+qwen3_omni_30b_a3b_captioner = ModelMeta(
+    loader=QwenOmniWrapper,
+    loader_kwargs=dict(
+        model_class="Qwen3OmniMoeThinkerForConditionalGeneration",
+    ),
```
Comment (AdnanElAssadi) [inline mteb/models/model_implementations/qwen_omni_lm.py]:
```suggestion

```

Comment (AdnanElAssadi) [reply mteb/models/model_implementations/qwen_omni_lm.py]: You don't seem to provide `loader_kwargs` else where.

**Comment by isaac-chung** [comment]
I couldn't find AudioSet but there's MusicCaps in table4 of the [technical report](https://arxiv.org/pdf/2503.20215). Might help confirm the implementation better?

**Comment by Samoed** [comment]
I got for `Qwen2.5-Omni-7B`, but I don't think we can compare these scores with paper, because they're doing STT
- `MusicCapsA2TRetrieval` `0.00117` 
- `MusicCapsT2ARetrieval` `0.00093`

To compare `Qwen2-Audio-7B` have `0.0014`, `0.0021`

**Review by AdnanElAssadi56 — APPROVED** [review]

File: mteb/models/model_implementations/qwen_omni_lm.py [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+                videos=video_inputs,
+                padding=True,
+                return_tensors="pt",
+                use_audio_in_video=False,
+            ).to(self.device)
+
+            outputs = self.model(
+                **model_inputs, output_hidden_states=True, return_dict=True
+            )
+            embeddings = outputs.hidden_states[-1][:, -1]
```
Comment (KennethEnevoldsen) [inline mteb/models/model_implementations/qwen_omni_lm.py]: I am not quite sure what is being selected here - last lidden states and teh
```suggestion
            embeddings = outputs.hidden_states[-1][:, -1] # select last hidden state ([-1]) and last token position ([:, -1]).
```

File: mteb/models/model_implementations/qwen_omni_lm.py [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+            embeddings = embeddings.to(torch.float32)
```
Comment (KennethEnevoldsen) [inline mteb/models/model_implementations/qwen_omni_lm.py]: why do we need this conversion? (required for normalization?)

File: mteb/models/model_implementations/qwen_omni_lm.py [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+            embeddings = torch.nn.functional.normalize(embeddings, p=2, dim=-1)
+            all_embeddings.append(embeddings.cpu())
+
+        return torch.cat(all_embeddings, dim=0).numpy()
```
Comment (KennethEnevoldsen) [inline mteb/models/model_implementations/qwen_omni_lm.py]:
```suggestion
        return torch.cat(all_embeddings, dim=0)
```

File: mteb/models/model_implementations/qwen_omni_lm.py [hunk 5]
```diff
@@ -0,0 +1,379 @@
...
+
+if TYPE_CHECKING:
+    from torch.utils.data import DataLoader
+
+    from mteb.abstasks.task_metadata import TaskMetadata
+    from mteb.types import Array, BatchedInput, PromptType
+
+
+class QwenOmniWrapper(AbsEncoder):
+    """Wrapper for Qwen Omni models supporting audio and images."""
```
Comment (KennethEnevoldsen) [inline mteb/models/model_implementations/qwen_omni_lm.py]: Should we add a brief description on how this is converted an embedding - last token, last hidden state

File: mteb/models/model_implementations/qwen_omni_lm.py (hunk 3 above)
Comment (Samoed) [reply mteb/models/model_implementations/qwen_omni_lm.py]: Similarly implemented in LCO, but I can try to run without it https://github.com/embeddings-benchmark/mteb/blob/7f1fa2ba605649d6fc913c6fc5eabc9e6f014bb1/mteb/models/model_implementations/lco_embedding_models.py#L127-L129

**Comment by AdnanElAssadi** [comment]
Hi @Samoed,

Can you add video here then we can merge?

**Comment by Samoed** [comment]
Yes, will add video
